pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -220,7 +220,9 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundHTTPRouteConfig(env *m
 // a proxy node
 func generateVirtualHostDomains(service *model.Service, port int, node *model.Proxy) []string {
 	domains := []string{string(service.Hostname), fmt.Sprintf("%s:%d", service.Hostname, port)}
```

Comment:
line 222: this is possibly why there is a duplicate as Service.Hostname = ${POD_NAMESPACE}.svc.local.cluster.